server/src/main/java/server/bazel/bazelWorkspaceAPI/WorkspaceAPI.java

```diff
@@ -0,0 +1,213 @@
+        }
+        this.workspaceTree = workspaceTree;
+    }
+
+    /**
+     *
+     * @param workspaceTree an initialized workspace Tree object
+     * @return The current WorkspaceAPI object with the newly set workspaceTree
+     * @throws WorkspaceAPIException if WorkspaceTree has no root
+     */
+    public WorkspaceAPI setWorkspace(WorkspaceTree workspaceTree) throws WorkspaceAPIException {
+        if (workspaceTree.getRoot() == null){
+            throw new WorkspaceAPIException("Workspace root cannot be null");
+        }
+        this.workspaceTree = workspaceTree;
+        return this;
+    }
+
+    // Publicly available methods
+    /**
+     *
+     * @param currentPath The pathway represented as a String to the package where you want to look for other possible packages
+     *          expected format "//path/to/"
+     * @return A list of possible paths as Strings
+     *          expected output: list = {//path/to/available, ...}
+     * @throws WorkspaceAPIException if path is invalid
+     */
+    public List<String> findPossibleCompletionsForPath(String currentPath) throws WorkspaceAPIException {
+        ArrayList<String> allPossiblePaths = new ArrayList<>();
+        // This will throw an exception if the path is invalid
+        List<Package> allPossiblePackages = findNodeOfGivenPackagePath(PathType.PACKAGE_PATH, currentPath).getAllPackagesOfChildren();
+
+        for(Package childPackage: allPossiblePackages){
+            // I might need to change the / to change based on the Operating system?
+            String sb = currentPath +
+                    childPackage.getPackageName();
+            allPossiblePaths.add(sb);
+        }
+        return allPossiblePaths;
+    }
+
+    /**
+     *
+     * @param pathToPackage The path represented as a string to the package that contains possible Build Targets
+     *          expected format "//path/to:"
+     *
+     * @return A List of Paths, represented by a string, of each possible build target
+     *          expected output: list = {//path/to/target:target, ....}
+     * @throws WorkspaceAPIException if the pathToPackage is an invalid path within the given Workspace
+     */
+    public List<String> findPossibleTargetsForPath(String pathToPackage) throws WorkspaceAPIException {
+        ArrayList<String> allPossibleTargets = new ArrayList<>();
+
+        Package packageFromPath =  findNodeOfGivenPackagePath(PathType.BUILT_TARGET_PATH,pathToPackage).getValue();
+        for(BuildTarget targets: packageFromPath.getBuildTargets()){
+            allPossibleTargets.add(targets.getPathWithTarget());
+        }
+        return  allPossibleTargets;
+    }
+
+    /**
+     *
+     * @param targetPath The path, represented by a String, to the given build target
+     *          expected format "//path/to:targetA"
+     * @return true if the build target is stored in the workspace tree
+     */
+    public boolean isValidTarget(String targetPath){
```

Comment:
We may want to abstract away strings that resemble `//path/to:targetA` bazel paths into their own domain object. They'll be reused a lot

Reply:
The `BuildTarget` class has this information.

Reply:
The BuildTarget class has this information.

Reply:
Abstracted out to BuildTarget object